sourcegraph/sourcegraph
# Prioritise SSO sign-in if exists

Closes https://github.com/sourcegraph/sourcegraph/issues/47444.

[Demo video](https://www.loom.com/share/1a27acf4bc924311a1ced0ae6772fee2)

## Screenshots
| Description/Scenario | Before | After |
| --: | :--: | :-- |
| When multiple SSO options exist | <img width="1728" alt="image" src="https://user-images.githubusercontent.com/6717049/216393098-f911448c-9392-43cc-821d-c29609796cfe.png"> | <img width="1728" alt="image" src="https://user-images.githubusercontent.com/6717049/216607670-df12e412-34ef-4bb0-8ead-ad96bfc0104c.png"> |
| When clicked on "Continue with Email" | None (scenario doesn't exist) | <img width="1728" alt="image" src="https://user-images.githubusercontent.com/6717049/216392355-2eb607ae-0044-4a4a-b4d0-b385f5924cab.png">|
| When only built-in auth option | <img width="1728" alt="image" src="https://user-images.githubusercontent.com/6717049/216393202-d60117cd-4028-4233-b8e1-49f0d9ddc638.png"> | <img width="1728" alt="image" src="https://user-images.githubusercontent.com/6717049/216392573-56d8f035-ba0e-48ad-9add-21951f9e9839.png"> |
| When built-in auth disabled | <img width="1728" alt="image" src="https://user-images.githubusercontent.com/6717049/216393286-ebe6e2a6-fd1d-4b65-93b7-b6de73f6e37c.png"> | <img width="1728" alt="image" src="https://user-images.githubusercontent.com/6717049/216607782-24e15958-4b51-4f51-9602-da817eaed858.png"> |


## Test plan
- Run locally in different modes (enterprise, dotcom, oss)
- Update siteAdmin > `auth.providers` to different variations (multiple sso, only built-in, no built-in, but multiple sso) and check http://localhost:3080/sign-in page


## App preview:

- [Web](https://sg-web-erzhtor-prioritize-sso-signin-if.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
